release/models/platform/openconfig-platform.yang

```diff
@@ -542,9 +548,13 @@ module openconfig-platform {
     leaf model-name {
       type string;
       description
-        "A human readable string describing the model of a component.
-        This string is optional and should only be populated if part-no
-        is also populated.";
+        "The model of a component. This should be the orderable part
+        number when the component is a FRU (field replaceable unit).
```

Comment:
Here's an example (this is a hypothetical example, but I assure you this issue exists in production):

```
/components/component[name="fabric1"]/state/part-no = "123-4567890"
/components/component[name="fabric1"]/state/model-name =  "FAB-100-M2"
```

In this example, the part-no is referencing some part+revision number which is internal to the vendor hardware revision.  This number might change even when the SKU/orderable part does not change (for example due to minor hardware revisions that do not affect the feature set).   The model-name is a "marketing sku" which is familiar to humans, appears in parts catalogs and can be ordered.

Not all vendors do this.  Some vendors only expose a "marketing sku" which is orderable.  In this case the leaves are redundant:
```
/components/component[name="fabric1"]/state/part-no = "FAB-100-M2"
/components/component[name="fabric1"]/state/model-name =  "FAB-100-M2"
```

Reply:
For context, I also plan on updating the type of this leaf from a string to an identity (https://github.com/openconfig/public/pull/1161) so that everyone is on the same page on which model names are expected.

As another example, this Nokia 7250 IXR-e  chassis has a `state/description` of "250 IXR-10e" and a `state/part-no` of "3HE16997AARB01" (and no `state/model-name`). I would like for us to add some form of this "250-ixr-10e"  to the model-name identity so that I could use `state/model-name` path to determine which chassis this is (rather than parsing `state/description`). 

In summary, I would like `state/model-name` to solve 2 problems:

1. Provide a consistent location for the model name (some vendors use description, some use part-no, etc.) so that I only have to use one path.
2. Define what values we expect to see (using the identity) so that I don't have to parse any values.

Reply:
I understand the idea, I just find it confusing to call something a part number and then proceed to say that it is not, in fact, a part number.

```In this example, the part-no is referencing some part+revision number which is internal to the vendor hardware revision. This number might change even when the SKU/orderable part does not change (for example due to minor hardware revisions that do not affect the feature set).```

That is true, however, only the revision changes. In our case, for example, the orderable part is the part number minus the revision (`3HE16997AA` in Sydney's example).

I'm not debating the leaf's overall meaning; I just think the new description is less clear than before.

Reply:
Ok yes I see your point. I have updated the description so that it doesn't say that model-name is the only one that could be searched online and I no longer call it a part number.

Do you have any other suggestions on what I could add to help explain the difference between part-no and model-name? I understand that both could be used to identify a component and potentially to search one up, but am not sure how else to explain that part-no is more often a "random" string whereas model-name is more of a "name" that has more meaning to someone who is reading it without a lot of background.

Reply:
I think the latest description resolves the ambiguity for model-name.